geomstats/_backend/pytorch/__init__.py

```diff
@@ -296,8 +296,15 @@ def einsum(*args, **kwargs):
         if len(input_str_list) > 2:
             raise NotImplementedError(
                 'Ellipsis support not implemented for >2 input tensors')
+        ndims = [len(input_str[3:]) for input_str in input_str_list]
```

Comment:
L286:
`einsum_list = einsum_str.split('->')`
Would the rest of the code work if there are no '->' in the string, as is sometimes the case?

Reply:
L292: `input_str[:3] == '...'`
ellipses could actually be in the middle of the input str (see e.g. exponential_barycenter l.65, this was a way to deal with various point types)

Reply:
Yes, I addressed this in PR #568 and put a TODO, since it's enhancement.